org.opentosca.container.api/src/org/opentosca/container/api/controller/CsarController.java

```diff
@@ -154,15 +151,21 @@ public DirectoryController getContent(@PathParam("csar") final String id) {
     @Consumes(MediaType.MULTIPART_FORM_DATA)
     @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
     @ApiOperation(hidden = true, value = "")
-    public Response uploadCsar(@FormDataParam(value = "file") final InputStream is,
+    public Response uploadCsar(@FormDataParam(value = "enrichment") final String applyEnrichment,
+                               @FormDataParam(value = "file") final InputStream is,
                                @FormDataParam("file") final FormDataContentDisposition file) {
 
         if (is == null || file == null) {
             return Response.status(Status.BAD_REQUEST).build();
         }
 
         logger.info("Uploading new CSAR file \"{}\", size {}", file.getFileName(), file.getSize());
-        return handleCsarUpload(file.getFileName(), is);
+
+        if (applyEnrichment != null) {
+            return handleCsarUpload(file.getFileName(), is, applyEnrichment);
+        } else {
+            return handleCsarUpload(file.getFileName(), is);
```

Comment:
null-handling sollte in `handleCsarUpload` geschehen, dadurch wird eine zusätzliche überladung vermieden